Spinner/Spinner.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Provides Agate-themed indeterminate progress indicator (spinner) components and behaviors.
+ *
+ * Used for indicating to the user that something is busy and interaction is temporarily suspended.
+ *
+ * @example
+ * <Spinner color="light" />
+ *
+ * @module agate/Spinner
+ * @exports Spinner
+ * @exports SpinnerBase
+ * @exports SpinnerDecorator
+ */
+import kind from '@enact/core/kind';
+import PropTypes from 'prop-types';
+import Pure from '@enact/ui/internal/Pure';
+import compose from 'ramda/src/compose';
+import React from 'react';
+import UiSpinnerBase from '@enact/ui/Spinner';
+
+import Skinnable from '../Skinnable';
+
+import componentCss from './Spinner.module.less';
+
+/**
+ * A component that shows spinning balls, with optional text as children.
+ *
+ * @class SpinnerCore
+ * @memberof agate/Spinner
+ * @ui
+ * @private
+ */
+const SpinnerCore = kind({
+	name: 'SpinnerCore',
+
+	propTypes: {
+		css: PropTypes.object
+	},
+
+	styles: {
+		css: componentCss
+	},
+
+	render: ({css, ...rest}) => (
+		<div aria-label="Loading" aria-live="off" role="alert" {...rest}>
+			<div className={css.bg}>
+				<div className={css.fan1} />
```

Comment:
It may be a good idea to have these nodes be generated via a `map`. That way, a computed property could decide the about of elements to create and we could easily support spinners with different amounts of nodes inside. Something like: 
```
computed: {
    spinnerNodes: ({type}) => (type === 'searching' ? 4 : 12)
}
```